server/sonar-server/src/main/java/org/sonar/server/issue/ws/IssueJsonWriter.java

```diff
@@ -42,15 +45,46 @@
 import org.sonar.core.component.ComponentDto;
 import org.sonar.markdown.Markdown;
 import org.sonar.server.user.UserSession;
+import org.sonar.server.ws.JsonWriterUtils;
+
+import static org.sonar.server.ws.JsonWriterUtils.writeIfNeeded;
 
 public class IssueJsonWriter {
 
-  public static final String ACTIONS_EXTRA_FIELD = "actions";
-  public static final String TRANSITIONS_EXTRA_FIELD = "transitions";
-  public static final String ACTION_PLAN_NAME_EXTRA_FIELD = "actionPlanName";
+  private static final String FIELD_KEY = "key";
+  private static final String FIELD_COMPONENT = "component";
+  private static final String FIELD_COMPONENT_ID = "componentId";
+  private static final String FIELD_PROJECT = "project";
+  private static final String FIELD_SUB_PROJECT = "subProject";
+  private static final String FIELD_RULE = "rule";
+  private static final String FIELD_STATUS = "status";
+  private static final String FIELD_RESOLUTION = "resolution";
+  private static final String FIELD_AUTHOR = "author";
+  private static final String FIELD_REPORTER = "reporter";
+  private static final String FIELD_ASSIGNEE = "assignee";
+  private static final String FIELD_DEBT = "debt";
+  private static final String FIELD_LINE = "line";
+  private static final String FIELD_MESSAGE = "message";
+  private static final String FIELD_SEVERITY = "severity";
+  private static final String FIELD_ACTION_PLAN = "actionPlan";
+  private static final String FIELD_CREATION_DATE = "creationDate";
+  private static final String FIELD_UPDATE_DATE = "updateDate";
+  private static final String FIELD_CLOSE_DATE = "closeDate";
+  private static final String FIELD_TAGS = "tags";
+  private static final String FIELD_COMMENTS = "comments";
+  private static final String FIELD_ATTRIBUTES = "attr";
+  public static final String FIELD_ACTIONS = "actions";
+  public static final String FIELD_TRANSITIONS = "transitions";
+  public static final String FIELD_ACTION_PLAN_NAME = "actionPlanName";
 
   public static final Set<String> EXTRA_FIELDS = ImmutableSet.of(
-    ACTIONS_EXTRA_FIELD, TRANSITIONS_EXTRA_FIELD, ACTION_PLAN_NAME_EXTRA_FIELD);
+    FIELD_ACTIONS, FIELD_TRANSITIONS, FIELD_ACTION_PLAN_NAME);
+
+  public static final Set<String> SELECTABLE_FIELDS = ImmutableSet.of(FIELD_COMPONENT, FIELD_PROJECT, FIELD_SUB_PROJECT, FIELD_RULE, FIELD_STATUS, FIELD_RESOLUTION, FIELD_AUTHOR,
+    FIELD_REPORTER, FIELD_ASSIGNEE, FIELD_DEBT, FIELD_LINE, FIELD_MESSAGE, FIELD_SEVERITY, FIELD_ACTION_PLAN, FIELD_CREATION_DATE, FIELD_UPDATE_DATE, FIELD_CLOSE_DATE,
+    FIELD_COMPONENT_ID, FIELD_TAGS, FIELD_COMMENTS, FIELD_ATTRIBUTES, FIELD_ACTIONS, FIELD_TRANSITIONS, FIELD_ACTION_PLAN_NAME);
+
+  private static final List<String> SELECTABLE_MINUS_EXTRAS = ImmutableList.copyOf(Sets.difference(SELECTABLE_FIELDS, EXTRA_FIELDS));
 
   private final I18n i18n;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "i18n" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1068)